### PR TITLE
Porting and fixing tests

### PR DESCRIPTION
**Pull request by frenzymadness**

### CODE REVIEW

**Comment by hroncok** [comment]
Does this obsolete https://github.com/fedora-python/python-ethtool/pull/18?

File: python-ethtool/include/py3c/compat.h [hunk 1]
```diff
@@ -0,0 +1,139 @@
+/* Copyright (c) 2015, Red Hat, Inc. and/or its affiliates
+ * Licensed under the MIT license; see py3c.h
```
Comment (hroncok) [inline python-ethtool/include/py3c/compat.h]: Be careful with this. It's referencing a file that's not here.

**Comment by hroncok** [comment]
BTW include/py3c/compat.h is not in sdist a tox cannot even attempt to compile this

**Comment by frenzymadness** [comment]
I am gonna cherry-pick your commits from #18 and merge these PRs. License and MANIFEST file fixed.

**Comment by hroncok** [comment]
I've seen a commit that made "Bring some CPython 2 helper code so we can run this" redundant by calling `unittest.main()` instead. Was it removed?

**Comment by frenzymadness** [comment]
Yes, I removed redundant commits during the merge.

**Comment by hroncok** [comment]
Did it not work? It was way better than my commit.

**Comment by frenzymadness** [comment]
Heh, I thought that your code is more complex and better.

I don't see any use case nor usage of soak tests so what about deleting this functionality?

**Comment by frenzymadness** [comment]
Seems like I **won** :smile:

File: .travis.yml [hunk 2]
```diff
@@ -1,4 +1,5 @@
 language: python
+sudo: required # Required to run in full VM instead of container
```
Comment (hroncok) [inline .travis.yml]: I this still required? Have you tried without?

Comment (frenzymadness) [inline .travis.yml]: Tested without `sudo`, Fixed.

**Comment by stratakis** [comment]
wow you nailed it guys

**Comment by hroncok** [comment]
Still WIP?

**Comment by frenzymadness** [comment]
@hroncok No, thanks.

**Pull request merged**